lib/services/admin.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:math';
+
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+
+class Admin {
+  final User? user;
+  String class_pin;
+  final bool mock;
+
+  Admin({this.user, this.class_pin = "0000", this.mock = false});
+
+  Future<String?> getCurrentUserId() async {
+    //var user = await auth.currentUser;
+    if (!mock) {
```

Comment:
we can not have al the methods use as isMock boolean. That is why we use mocking. Please refactor this method and do appropriate changes in the tests to properly mock a user and for this method to return a fake userId not bcs it's hardcoded in it and it knows it is operating as a test but because we pass it a fake FirebaseAuth object and a fake User object that has this id